requirements/dev/dev.in

```diff
@@ -2,22 +2,22 @@
 -c ../base/base.txt
 
 pyyaml==6.0.1
-black==22.6.0
+black==24.4.2
 isort
 ipython
 jupyterlab
 
 # deploy
-ansible==7.6.0
+ansible==9.5.1
 invoke-kubesae==0.1.0
 
 # AWS tools
-awscli==1.29.4
+awscli==1.32.104
 awslogs
 Jinja2==3.0.3
-openshift==0.12
+openshift==0.13.2
 kubernetes==12.0.0
-kubernetes-validate~=1.25.0
+kubernetes-validate~=1.29.1
```

Comment:
I see the following error running the `deploy-cluster.yml` playbook:

```
TASK [caktus.k8s-web-cluster : Grant access to IAM users if aws] *************************************************************************************************************
fatal: [production]: FAILED! => changed=false 
  msg: Failed to import the required Python library (kubernetes-validate) on rana.lan's Python /Users/copelco/projects/philly-hip/.direnv/python-3.10/bin/python3.10. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter
```

kubernetes-validate appears to be installed, but it does throw an exception when importing it:

```python
> python                             
Python 3.10.11 (v3.10.11:7d4cc5aa85, Apr  4 2023, 19:05:19) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import kubernetes_validate
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/copelco/projects/philly-hip/.direnv/python-3.10/lib/python3.10/site-packages/kubernetes_validate/__init__.py", line 1, in <module>
    from .utils import latest_version, validate, validate_file, validate_resource
  File "/Users/copelco/projects/philly-hip/.direnv/python-3.10/lib/python3.10/site-packages/kubernetes_validate/utils.py", line 14, in <module>
    from referencing import Registry, Resource
  File "/Users/copelco/projects/philly-hip/.direnv/python-3.10/lib/python3.10/site-packages/referencing/__init__.py", line 1, in <module>
    from referencing._core import (  # noqa: F401
  File "/Users/copelco/projects/philly-hip/.direnv/python-3.10/lib/python3.10/site-packages/referencing/_core.py", line 6, in <module>
    from attrs import evolve, field
ModuleNotFoundError: No module named 'attrs'
```

This is confusing to me, because pip freeze reports it's installed:

```shell
> pip freeze | grep attrs                                                                                      
attrs==20.3.0
```

Upgrading the dev requirements ended up working for me:

```sh
pip-compile --output-file=requirements/dev/dev.txt requirements/dev/dev.in --upgrade
make setup
```

But then the tests failed to run (maybe related to `wagtail-factories`?), so it may require additional investigation.

Reply:
Interesting, the playbook works/runs for me.